Lib/sysconfig.py

```diff
@@ -449,18 +448,15 @@ def parse_config_h(fp, vars=None):
         line = fp.readline()
         if not line:
             break
-        m = define_rx.match(line)
-        if m:
+        if (m := define_rx.match(line)):
             n, v = m.group(1, 2)
             try:
                 v = int(v)
             except ValueError:
                 pass
             vars[n] = v
-        else:
-            m = undef_rx.match(line)
-            if m:
-                vars[m.group(1)] = 0
+        elif (m := undef_rx.match(line)):
+            vars[m.group(1)] = 0
```

Comment:
This one looks like an improvement to me, because otherwise I smell a bug ... what if it went down the else case, but then the 2nd if wasn't true?  Is that case handled?

Now I still smell the potential bug, but with an elif, it is a lot easier to explain what I'm worried about, as a simple "What if neither pattern matches?"